tests/unit/onyxClearWebStorageTest.js

```diff
@@ -50,26 +50,6 @@ describe('Set data while storage is clearing', () => {
         return Onyx.clear();
     });
 
-    it('should persist the value of Onyx.merge when called between the cache and storage clearing', () => {
```

Comment:
@neil-marcellini I saw you added these tests, and as far as I can tell, they are no longer necessary because the behavior of `clear()` has been changed. Can you please take a look at this and make sure that I am not introducing regressions that I'm not aware of? 

Basically, the way I read this is that the problem only happened because we were done something like this:
```
Onyx.clear();
Onyx.set(whatever);
```

My PR here changes this behavior so that no one should ever be doing that anymore. It should now become:

```
Onyx.clear(keysToPreserve);
```
So that's why I think it's OK to remove these tests since they no longer test a valid use case.

Reply:
These tests make sure that the following flow works for keys with a default key state, like the session.
1. Sign out the user with Onyx.clear()
2. Sign in a new user (before Onyx.clear finishes), and merge new data on the session key
3. Ensure that Onyx.clear doesn't set a default key state on the session key and wipe out the value saved in step 2.

I think it might still be valuable to test calling `Onyx.clear()` and then `Onyx.merge` on a key with a default state.

Reply:
OK, thanks! My point is that this...

> Sign in a new user (before Onyx.clear finishes),

...isn't possible to do. `Onyx.clear()` takes less that 200ms to complete now on an account with 4000 reports, as well as we are no longer optimistically redirecting to the sign-on page. We are waiting until `Onyx.clear()` is finished before the redirect happens.

If there is a valid use case of calling `clear -> merge`, I could see having a test, but I don't know if any time that would ever be a valid case. Do you have a specific one in mind? If that does exist in the code, it should be rewritten like I showed above.

Reply:
> `Onyx.clear()` takes less that 200ms to complete now on an account with 4000 reports,

Ah ok nice. If that's the case for all platforms then I agree that we don't need this test any more 👍.



> we are no longer optimistically redirecting to the sign-on page. We are waiting until `Onyx.clear()` is finished before the redirect happens.

I don't think we are based on the code [here](https://github.com/Expensify/App/blob/14eee462c7da856fe5ef38a5e5343d655e9b62b4/src/libs/actions/SignInRedirect.js#L45-L46). As soon as the session key is cleared we are signed out, and that can happen before Onyx finishes clearing.

Reply:
> As soon as the session key is cleared we are signed out, and that can happen before Onyx finishes clearing.

Ah yes, you're technically right. However, the session key is removed at the same time as all other keys now (because it is once again using `LocalForage.clear()` under the hood rather than using a `multiSet()` or `multiRemove()`).